solver.py

```diff
@@ -211,18 +211,16 @@ def solve(self, report=True):
                 findFreeParams=False,
             )
 
-            # NOTE: For some reason solve() might return undocumented values,
-            # Clamp result value to 4
-            if retval > 3:
+            if retval > 5:
                 logger.debug("Solver returned undocumented value: {}".format(retval))
-                retval = 4
+
             self.result = bpyEnum(solver_state_items, index=retval)
 
             if report and sketch:
                 sketch.solver_state = self.result.index
                 sketch.dof = self.solvesys.Dof
 
-            if retval != 0:
+            if retval != 0 and retval != 5:
```

Comment:
Why are redundant constraints not an error state?

Reply:
Two reasons basically:
1. That is the way the C++ library used actually handles these things in their upper layers (note that CAD Sketcher uses lower layers).
2. From what I could gather there seems to be no (easy) way of detecting these redundant constraints.

Reply:
Now that I tested it, it is really nice to only have it as a warning and not an error, as the solver seems to ignore duplicates/redundant constraints :)